src/lightning/pytorch/callbacks/rich_model_summary.py

```diff
@@ -52,21 +52,24 @@ class RichModelSummary(ModelSummary):
     Args:
         max_depth: The maximum depth of layer nesting that the summary will include. A value of 0 turns the
             layer summary off.
+        header_style: Style of the table header.
 
     Raises:
         ModuleNotFoundError:
             If required `rich` package is not installed on the device.
     """
 
-    def __init__(self, max_depth: int = 1) -> None:
+    def __init__(self, max_depth: int = 1, header_style: str = "bold magenta") -> None:
         if not _RICH_AVAILABLE:
             raise ModuleNotFoundError(
                 "`RichModelSummary` requires `rich` to be installed. Install it by running `pip install -U rich`."
             )
         super().__init__(max_depth)
 
-    @staticmethod
```

Comment:
The base interface defines `summarize` as a staticmethod. So doing this is not possible as it would be a breaking change, and also breaks typing (see the failing mypy job)

Supporting this would need to be done by adding `summarize_kwargs: Dict[str, Any]` to the base interface, where they get passed to `self.summarize(..., **self._summarize_kwargs)` in `on_fit_start` of the parent.

Reply:
Fixed it, Mypy passes now